### PR TITLE
Add Snowflake Extractor Resource

### PR DESCRIPTION
**Pull request by wtran63**
Changes include:

- Added extractor resource to enable extractors being handled by terraform. 
- Snowflake credentials are exactly the same as found on the writer resource so a centralized schema was created to avoid duplication of structs
- Added ID Generation for extractor tables

### CODE REVIEW

File: plugin/providers/keboola/resource_keboola_snowflake_extractor.go [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+	"fmt"
+	"log"
+	"net/url"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/plmwong/terraform-provider-keboola/plugin/providers/keboola/buffer"
+)
+
+
+type SnowflakeExtractor struct {
```
Comment (plmwong) [inline plugin/providers/keboola/resource_keboola_snowflake_extractor.go:14]: The spacing/tabbing seems to be all over the show. I think VSCode and GoLand both can automatically align everything.

File: plugin/providers/keboola/resource_keboola_snowflake_extractor.go [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+	return nil
+}
+
+func resourceKeboolaSnowflakeExtractorRead(d *schema.ResourceData, meta interface{}) error {
+	log.Println("[INFO] Reading Snowflake Extractor from Keboola.")
+
+	client := meta.(*KBCClient)
+	getSnowflakeExtractorResponse, err := client.GetFromStorage(fmt.Sprintf("storage/components/keboola.ex-db-snowflake/configs/%s", d.Id()))
+
+	if d.Id() == "" {
```
Comment (plmwong) [inline plugin/providers/keboola/resource_keboola_snowflake_extractor.go]: Wondering if this should be happening before the above `GetFromStorage` call? Like, if the `Id` is missing, then short circuit and bail straight away without calling any API?

File: plugin/providers/keboola/resource_keboola_snowflake_extractor_tables.go [hunk 3]
```diff
@@ -0,0 +1,401 @@
...
+	if err != nil {
+		return err
+	}
+
+	var tables []map[string]interface{}
+
+	for _, extractorTable := range SnowflakeExtractor.Configuration.Parameters.Tables {
+
+		tableDetails := map[string]interface{}{
+			"id":				extractorTable.ID,
```
Comment (plmwong) [inline plugin/providers/keboola/resource_keboola_snowflake_extractor_tables.go]: These are all indented weird as well.

File: plugin/providers/keboola/resource_keboola_snowflake_extractor_tables.go [hunk 4]
```diff
@@ -0,0 +1,401 @@
...
+	extractorTables := make([]SnowflakeExtractorTable, 0, len(tables))
+
+	distinctNames := make(map[string]bool)
+	distinctIds := make(map[int]bool)
+
+	for _, table :=  range tables {
+		config := table.(map[string]interface{})
+
+		extractorTable := SnowflakeExtractorTable{
+			Name: 			config["name"].(string),
```
Comment (plmwong) [inline plugin/providers/keboola/resource_keboola_snowflake_extractor_tables.go]: Here as well. Indentation. Weird.

File: plugin/providers/keboola/resource_keboola_snowflake_extractor_test.go [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccSnowflakeExtractor_Basic(t *testing.T) {
```
Comment (plmwong) [inline plugin/providers/keboola/resource_keboola_snowflake_extractor_test.go:11]: Are we also missing tests for the `tables` resource?

File: plugin/providers/keboola/resource_keboola_snowflake_extractor.go (hunk 1 above)
Comment (wtran63) [reply plugin/providers/keboola/resource_keboola_snowflake_extractor.go:14]: Updated. Fixed.

File: plugin/providers/keboola/resource_keboola_snowflake_extractor_tables.go (hunk 3 above)
Comment (wtran63) [reply plugin/providers/keboola/resource_keboola_snowflake_extractor_tables.go]: Updated. Fixed.

File: plugin/providers/keboola/resource_keboola_snowflake_extractor_tables.go (hunk 4 above)
Comment (wtran63) [reply plugin/providers/keboola/resource_keboola_snowflake_extractor_tables.go]: Updated. Fixed. Indents converted to spaces

File: plugin/providers/keboola/resource_keboola_snowflake_extractor.go (hunk 2 above)
Comment (wtran63) [reply plugin/providers/keboola/resource_keboola_snowflake_extractor.go]: This have been moved above the API call.

File: plugin/providers/keboola/resource_keboola_snowflake_extractor_test.go (hunk 5 above)
Comment (wtran63) [reply plugin/providers/keboola/resource_keboola_snowflake_extractor_test.go:11]: Yes... The writer tables resource did not have tests as well, and I figured I could follow the same convention in terms of tests. 

Not saying this is a valid reason for not writing tests, but I could write some as well if you like

**Pull request merged**